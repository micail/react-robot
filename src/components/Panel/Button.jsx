import React from 'react';
import PropTypes from 'prop-types';

import './Button.scss';

const Button = (props) => {
  const name = Object.keys(props).toString();
  return (<button type="button" className="btn" id={`btn-${name}`}>{name.toUpperCase()}</button>);
};
export default Button;

Button.propTypes = {
};

Button.defaultProps = {
};
