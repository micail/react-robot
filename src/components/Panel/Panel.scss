.panel {
  margin-top: 1rem;

  .btn {
    margin: 0.2rem;
    min-width: 3rem;
  }
}