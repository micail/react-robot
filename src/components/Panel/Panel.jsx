import React from 'react';
import PropTypes from 'prop-types';

import Button from '../Button/Button';

import './Panel.scss';

const Panel = ({ commands }) => {
  return (
    <div className="panel">
      <Button name="move" func={commands.move} />
      <div>
        <Button name="left" func={commands.left} />
        <Button name="right" func={commands.right} />
      </div>
      <Button name="report" func={commands.report} />
      <Button name="place" func={commands.place} />
    </div>
  );
};

export default Panel;

Panel.propTypes = {
  commands: PropTypes.objectOf(PropTypes.func),
};

Panel.defaultProps = {
  commands: {
    move: () => { },
    left: () => { },
    right: () => { },
    report: () => { },
    place: () => { },
  },
};
