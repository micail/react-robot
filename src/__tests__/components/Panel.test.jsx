import React from 'react';
import { mount } from 'enzyme';

import Panel from '../../components/Panel/Panel';

describe('Panel component', () => {
  let component;
  beforeAll(() => {
    component = mount(<Panel />);
  });

  it('should render panel component', () => {
    expect(component).toMatchSnapshot();
  });
  it('should render move button', () => {
    const button = component.find('#btn-move');
    expect(button).toMatchSnapshot();
  });
  it('should render left button', () => {
    const button = component.find('#btn-left');
    expect(button.exists()).toBeTruthy();
  });
  it('should render right button', () => {
    const button = component.find('#btn-right');
    expect(button.exists()).toBeTruthy();
  });
  it('should render report button', () => {
    const button = component.find('#btn-report');
    expect(button.exists()).toBeTruthy();
  });
});
