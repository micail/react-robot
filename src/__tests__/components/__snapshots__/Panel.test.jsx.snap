// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Panel component should render move button 1`] = `
<button
  className="btn"
  id="btn-move"
  onClick={[Function]}
  type="button"
>
  MOVE
</button>
`;

exports[`Panel component should render panel component 1`] = `
<Panel
  commands={
    Object {
      "left": [Function],
      "move": [Function],
      "place": [Function],
      "report": [Function],
      "right": [Function],
    }
  }
>
  <div
    className="panel"
  >
    <Button
      func={[Function]}
      name="move"
    >
      <button
        className="btn"
        id="btn-move"
        onClick={[Function]}
        type="button"
      >
        MOVE
      </button>
    </Button>
    <div>
      <Button
        func={[Function]}
        name="left"
      >
        <button
          className="btn"
          id="btn-left"
          onClick={[Function]}
          type="button"
        >
          LEFT
        </button>
      </Button>
      <Button
        func={[Function]}
        name="right"
      >
        <button
          className="btn"
          id="btn-right"
          onClick={[Function]}
          type="button"
        >
          RIGHT
        </button>
      </Button>
    </div>
    <Button
      func={[Function]}
      name="report"
    >
      <button
        className="btn"
        id="btn-report"
        onClick={[Function]}
        type="button"
      >
        REPORT
      </button>
    </Button>
    <Button
      func={[Function]}
      name="place"
    >
      <button
        className="btn"
        id="btn-place"
        onClick={[Function]}
        type="button"
      >
        PLACE
      </button>
    </Button>
  </div>
</Panel>
`;
